notification/search.go

```diff
@@ -193,10 +222,13 @@ func (s *Store) TimeSeries(ctx context.Context, opts TimeSeriesOpts) ([]TimeSeri
 	}
 
 	data := &renderData{
-		SearchOptions:      opts.SearchOptions,
-		TimeSeries:         true,
-		TimeSeriesOrigin:   opts.TimeSeriesOrigin,
-		TimeSeriesInterval: opts.TimeSeriesInterval,
+		SearchOptions:        opts.SearchOptions,
+		TimeSeries:           true,
+		TimeSeriesOrigin:     opts.TimeSeriesOrigin,
+		TimeSeriesInterval:   opts.TimeSeriesInterval,
+		SegmentByService:     opts.SegmentBy == "service",
+		SegmentByUser:        opts.SegmentBy == "user",
+		SegmentByMessageType: opts.SegmentBy == "messageType",
```

Comment:
Should we make these a const/special type and link gql to them to ensure they match?

Reply:
Do you think the validation I uncommented will be sufficient? If not I can do something like that

Reply:
From an API standpoint, it should validate fine. Using constants just makes it cleaner to follow, adds type-checking, and potential refactoring made easier.